### PR TITLE
Ensemble Move short-task

### PR DESCRIPTION
**Pull request by amandarichardsonn**
The Ensemble class is moved to core/builder/ in this PR -> as_jobs is changed to build_jobs -> and documentation is updated.

### CODE REVIEW

File: smartsim/entity/__init__.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 from .application import Application
 from .dbnode import FSNode
 from .dbobject import *
-from .ensemble import Ensemble
+from .._core.builders.ensemble import Ensemble
```
Comment (mellis13) [inline smartsim/entity/__init__.py:30]: I think ``_core`` is mostly reserved for non-user infrastructure.  I think maybe ``builders`` should be in top-level SmartSim.  Thoughts?

**Review by mellis13 — CHANGES_REQUESTED** [review]

File: smartsim/entity/__init__.py (hunk 1 above)
Comment (amandarichardsonn) [reply smartsim/entity/__init__.py]: I believe that strategies.py might also need to be moved, am I correct on this?

Comment (amandarichardsonn) [reply smartsim/entity/__init__.py]: strategies moved

**Review by mellis13 — APPROVED** [review]
LGTM

**Pull request merged**